sdks/java/io/azure/src/test/java/org/apache/beam/sdk/io/azure/blobstore/AzureBlobStoreFileSystemTest.java

```diff
@@ -0,0 +1,352 @@
+  }
+
+  @Test
+  public void testGetScheme() {
+    assertEquals("azfs", azureBlobStoreFileSystem.getScheme());
+  }
+
+  @Test
+  public void testGlobTranslation() {
+    assertEquals("foo", AzureBlobStoreFileSystem.wildcardToRegexp("foo"));
+    assertEquals("fo[^/]*o", AzureBlobStoreFileSystem.wildcardToRegexp("fo*o"));
+    assertEquals("f[^/]*o\\.[^/]", AzureBlobStoreFileSystem.wildcardToRegexp("f*o.?"));
+    assertEquals("foo-[0-9][^/]*", AzureBlobStoreFileSystem.wildcardToRegexp("foo-[0-9]*"));
+    assertEquals("foo-[0-9].*", AzureBlobStoreFileSystem.wildcardToRegexp("foo-[0-9]**"));
+    assertEquals(".*foo", AzureBlobStoreFileSystem.wildcardToRegexp("**/*foo"));
+    assertEquals(".*foo", AzureBlobStoreFileSystem.wildcardToRegexp("**foo"));
+    assertEquals("foo/[^/]*", AzureBlobStoreFileSystem.wildcardToRegexp("foo/*"));
+    assertEquals("foo[^/]*", AzureBlobStoreFileSystem.wildcardToRegexp("foo*"));
+    assertEquals("foo/[^/]*/[^/]*/[^/]*", AzureBlobStoreFileSystem.wildcardToRegexp("foo/*/*/*"));
+    assertEquals("foo/[^/]*/.*", AzureBlobStoreFileSystem.wildcardToRegexp("foo/*/**"));
+    assertEquals("foo.*baz", AzureBlobStoreFileSystem.wildcardToRegexp("foo**baz"));
+  }
+
+  @Test
+  @SuppressWarnings("CheckReturnValue")
+  public void testCopy() throws IOException {
+    List<AzfsResourceId> src =
+        new ArrayList<>(
+            Arrays.asList(AzfsResourceId.fromComponents("account", "container", "from")));
+    List<AzfsResourceId> dest =
+        new ArrayList<>(Arrays.asList(AzfsResourceId.fromComponents("account", "container", "to")));
+    when(mockedBlobClient.exists()).thenReturn(true);
+    azureBlobStoreFileSystem.copy(src, dest);
+    verify(mockedBlobClient, times(1)).copyFromUrl(any(String.class));
+  }
+
+  @Test
+  public void testWriteAndRead() throws IOException {
+    azureBlobStoreFileSystem.getClient().createBlobContainer("testcontainer");
+
+    byte[] writtenArray = new byte[] {0};
+    ByteBuffer bb = ByteBuffer.allocate(writtenArray.length);
+    bb.put(writtenArray);
+
+    // First create an object and write data to it
+    AzfsResourceId path = AzfsResourceId.fromUri("azfs://account/testcontainer/foo/bar.txt");
+    WritableByteChannel writableByteChannel =
+        azureBlobStoreFileSystem.create(path, builder().setMimeType("application/text").build());
+    writableByteChannel.write(bb);
+    writableByteChannel.close();
+
+    // Now read the same object
+    ByteBuffer bb2 = ByteBuffer.allocate(writtenArray.length);
+    ReadableByteChannel open = azureBlobStoreFileSystem.open(path);
+    open.read(bb2);
+
+    // And compare the content with the one that was written
+    byte[] readArray = bb2.array();
+    assertArrayEquals(readArray, writtenArray);
+    open.close();
+  }
+
+  @Test
+  @Ignore
+  public void testGlobExpansion() throws IOException {
+    // TODO: Write this test with mocks - see GcsFileSystemTest
```

Comment:
Please file a JIRA issue to track this.

Reply:
Done: https://issues.apache.org/jira/browse/BEAM-10772